sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -471,6 +471,155 @@ void flash_ctrl_data_default_cfg_set(flash_ctrl_cfg_t cfg) {
+    mp_region_cfg = bitfield_field32_write(                                    \
+        mp_region_cfg,                                                         \
+        FLASH_CTRL_MP_REGION_CFG_##region##_ECC_EN_##region##_FIELD, cfg.ecc); \
+    mp_region_cfg = bitfield_field32_write(                                    \
+        mp_region_cfg,                                                         \
+        FLASH_CTRL_MP_REGION_CFG_##region##_SCRAMBLE_EN_##region##_FIELD,      \
+        cfg.scrambling);                                                       \
+    mp_region_cfg = bitfield_field32_write(                                    \
+        mp_region_cfg,                                                         \
+        FLASH_CTRL_MP_REGION_CFG_##region##_ERASE_EN_##region##_FIELD,         \
+        perms.erase);                                                          \
+    mp_region_cfg = bitfield_field32_write(                                    \
+        mp_region_cfg,                                                         \
+        FLASH_CTRL_MP_REGION_CFG_##region##_PROG_EN_##region##_FIELD,          \
+        perms.write);                                                          \
+    mp_region_cfg = bitfield_field32_write(                                    \
+        mp_region_cfg,                                                         \
+        FLASH_CTRL_MP_REGION_CFG_##region##_RD_EN_##region##_FIELD,            \
+        perms.read);                                                           \
+    mp_region_cfg = bitfield_field32_write(                                    \
+        mp_region_cfg,                                                         \
+        FLASH_CTRL_MP_REGION_CFG_##region##_EN_##region##_FIELD, en);          \
+    sec_mmio_write32(kBase + FLASH_CTRL_MP_REGION_CFG_##region##_REG_OFFSET,   \
+                     mp_region_cfg);                                           \
+    return;                                                                    \
+  }
+
+  switch (region) {
+    FLASH_CTRL_MP_REGIONS(FLASH_CTRL_MP_REGION_CFG_WRITE_)
+    default:
+      OT_UNREACHABLE();
+  }
+
+#undef FLASH_CTRL_MP_REGION_CFG_WRITE_
+}
+
+void flash_ctrl_data_region_protect(flash_ctrl_region_index_t region,
+                                    uint32_t page_offset, uint32_t num_pages,
+                                    flash_ctrl_perms_t perms) {
+  // Reset the region's configuration via the MP_REGION_CFG_${region} register.
+  // This temporarily disables memory protection for the region.
+  flash_ctrl_mp_region_cfg_reset(region);
+
+  // Set the region's bounds in the MP_REGION_${region} register.
+  flash_ctrl_mp_region_write(region, page_offset, num_pages);
+
+  // Read the DEFAULT_REGION register to determine default values for some
+  // fields of MP_REGION_CFG_${region}.
+  const flash_ctrl_cfg_t default_cfg = flash_ctrl_data_default_cfg_get();
```

Comment:
I don't think we want/need this here. the default setting will apply if no other region configs match and clients can handle if they need to.

Reply:
Hmm, I didn't entirely follow this comment. The purpose of `default_cfg` is to grab the default values for `he`, `ecc`, and `scrambling` when we construct the new value of the `MP_REGION_CFG_${region}` register.

Is there a better way to do this, or is it okay to just zero those bitfields out?

Reply:
Oh, I see what you mean. I think it would be clearer if we promote this to the function prototype, i.e. add a `flash_ctrl_cfg_t cfg` parameter instead of defaulting to `flash_ctrl_data_default_cfg_get()`s return value. Callers can then specify that as they want.

Reply:
Done! The diff was a little more than I expected because I had to update the mocks, but not very tricky.